kyo-prelude/shared/src/main/scala/kyo/Stream.scala

```diff
@@ -272,14 +272,16 @@ sealed abstract class Stream[V, -S]:
         discr: Stream.Dummy,
         frame: Frame
     ): Stream[V, S] =
-        Stream[V, S](ArrowEffect.handleState(tag, true, emit)(
```

Comment:
is this an unrelated refactoring? It's ok if it is, just to confirm

Reply:
Now that we're passing chunks directly I'm not using `takeWhile` or `collectWhile` anymore, and when I went back to make tests I wasn't able to reproduce the problematic behavior I ran into. I decided to revert to the old implementation.

I'm still keeping `Stream#collectWhile` in this PR, though, because it's useful to have in the API otherwise.